services/user/CommonSys/common/rpc.mjs

```diff
@@ -627,9 +632,45 @@ export async function initializeApplet(initializer = () => { }) {
 
     await import("/common/iframeResizer.contentWindow.js");
 
+    console.info("setting up applet event listeners popstate");
+    let currentHref = document.location.href;
+    const body = document.querySelector("body");
+    const observer = new MutationObserver((mutations) => {
+        mutations.forEach(() => {
+            if (currentHref !== document.location.href) {
+                console.info(">>> observed href mutation");
+                sendHistoryChange();
+                currentHref = document.location.href;
+            }
+        });
+    });
+    observer.observe(body, { childList: true, subtree: true });
+
+    // required to detect back button
+    window.onpopstate = (_event) => {
+        if (currentHref !== document.location.href) {
+            console.info(">>> observed popstate href mutation");
+            sendHistoryChange();
+            currentHref = document.location.href;
+        }
+    };
+
     await initializer();
 }
 
+const sendHistoryChange = () => {
+    const { pathname, href, search } = document.location;
+    console.info("sending change history message", {
+        href,
+        pathname,
+        search,
+    });
```

Comment:
We happy to lose this or shall keep it for now?

Reply:
I think for debug purposes it's a good idea to keep it for now! then if we dont see any anomalies we can go ahead and clean it up!